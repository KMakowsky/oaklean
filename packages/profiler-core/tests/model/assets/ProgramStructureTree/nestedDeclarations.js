export default class ExampleClass {
	memberFunction1() {
		function nestedFunction() {
			const arrowFunction = () => {
				function a() {}
				const b = () => {}
				const c = () => {}
				const d = () => {}
			}
		}
	}
}