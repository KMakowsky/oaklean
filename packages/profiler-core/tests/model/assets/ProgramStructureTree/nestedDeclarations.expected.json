{
  "id": 0,
  "type": "Root",
  "identifier": "{root}",
  "identifierType": "Name",
  "beginLoc": {
    "line": 1,
    "column": 0
  },
  "endLoc": {
    "line": 12,
    "column": 1
  },
  "children": {
    "{class:ExampleClass}": {
      "id": 1,
      "type": "ClassDeclaration",
      "identifier": "{class:ExampleClass}",
      "identifierType": "Name",
      "beginLoc": {
        "line": 1,
        "column": 0
      },
      "endLoc": {
        "line": 12,
        "column": 1
      },
      "children": {
        "{method:memberFunction1}": {
          "id": 2,
          "type": "MethodDefinition",
          "identifier": "{method:memberFunction1}",
          "identifierType": "Name",
          "beginLoc": {
            "line": 2,
            "column": 1
          },
          "endLoc": {
            "line": 11,
            "column": 2
          },
          "children": {
            "{function:nestedFunction}": {
              "id": 3,
              "type": "FunctionDeclaration",
              "identifier": "{function:nestedFunction}",
              "identifierType": "Name",
              "beginLoc": {
                "line": 3,
                "column": 2
              },
              "endLoc": {
                "line": 10,
                "column": 3
              },
              "children": {
                "{functionExpression:arrowFunction}": {
                  "id": 4,
                  "type": "ArrowFunctionExpression",
                  "identifier": "{functionExpression:arrowFunction}",
                  "identifierType": "Name",
                  "beginLoc": {
                    "line": 4,
                    "column": 25
                  },
                  "endLoc": {
                    "line": 9,
                    "column": 4
                  },
                  "children": {
                    "{function:a}": {
                      "id": 5,
                      "type": "FunctionDeclaration",
                      "identifier": "{function:a}",
                      "identifierType": "Name",
                      "beginLoc": {
                        "line": 5,
                        "column": 4
                      },
                      "endLoc": {
                        "line": 5,
                        "column": 19
                      },
                      "children": {}
                    },
                    "{functionExpression:b}": {
                      "id": 6,
                      "type": "ArrowFunctionExpression",
                      "identifier": "{functionExpression:b}",
                      "identifierType": "Name",
                      "beginLoc": {
                        "line": 6,
                        "column": 14
                      },
                      "endLoc": {
                        "line": 6,
                        "column": 22
                      },
                      "children": {}
                    },
                    "{functionExpression:c}": {
                      "id": 7,
                      "type": "ArrowFunctionExpression",
                      "identifier": "{functionExpression:c}",
                      "identifierType": "Name",
                      "beginLoc": {
                        "line": 7,
                        "column": 14
                      },
                      "endLoc": {
                        "line": 7,
                        "column": 22
                      },
                      "children": {}
                    },
                    "{functionExpression:d}": {
                      "id": 8,
                      "type": "ArrowFunctionExpression",
                      "identifier": "{functionExpression:d}",
                      "identifierType": "Name",
                      "beginLoc": {
                        "line": 8,
                        "column": 14
                      },
                      "endLoc": {
                        "line": 8,
                        "column": 22
                      },
                      "children": {}
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}